src/Val.cc

```diff
@@ -1453,6 +1563,9 @@ void TableVal::Init(TableTypePtr t, bool ordered) {
     else
         subnets = nullptr;
 
+    if ( table_type->IsPatternIndex() )
+        pattern_matcher = new detail::TablePatternMatcher(this, table_type->Yield());
+
```

Comment:
I realize that this is following the existing pattern, but moving to `unique_ptr` would be nice (maybe as a follow up covering the existing cases too).